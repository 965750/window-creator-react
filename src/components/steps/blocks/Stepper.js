import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import StepBox from './StepBox'
import styles from './Stepper.module.scss'

class Stepper extends Component {
    render() {
        return (
            <div className={`${styles.stepper} flex relative mx-auto mt-6`}>
                {this.props.steps.map(step => (
                    <StepBox
                        step={step}
                        active={step.id === this.props.activeStep}
                        key={step.id}
                    />
                ))}
                <div
                    className={`${styles.stepper__line} border-b border-mercury absolute z-10 left-half`}
                />
            </div>
        )
    }
}

const mapStateToProps = state => ({
    steps: state.theme.stepper.steps,
    activeStep: state.theme.stepper.active,
})

Stepper.propTypes = {
    steps: PropTypes.array.isRequired,
    activeStep: PropTypes.number.isRequired,
}

export default connect(mapStateToProps)(Stepper)
