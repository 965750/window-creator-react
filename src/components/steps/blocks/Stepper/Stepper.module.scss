.stepper {
    width: 450px;

    @media screen and (max-width: 768px) {
        width: 300px;
    }

    &__line {
        transform: translate(-50%, 0);
        top: 25px;
        width: 300px;

        @media screen and (max-width: 768px) {
            width: 250px;
        }
    }
}