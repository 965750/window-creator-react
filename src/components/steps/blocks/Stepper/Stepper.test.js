import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react';

import Stepper from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk'
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
    stepper: {
      steps: [
        {
          label: 'Step 1',
          title: 'Choose Door',
          id: 1,
        },
      ],
      active: 1
    }
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('Stepper', () => {
  let wrapper
  const initialProps = {}

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <Stepper {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
