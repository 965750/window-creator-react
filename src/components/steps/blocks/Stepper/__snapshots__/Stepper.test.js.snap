// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="stepper flex relative mx-auto mt-6"
  >
    <div
      class="w-1/3 text-center z-20"
    >
      <div
        class="circle bg-gullGray cursor-pointer rounded-full bg-botticelli relative mx-auto"
        data-testid="stepBoxCircle"
      >
        <div
          class="circle__inside bg-white top-half left-half absolute rounded-full"
          data-testid="stepBoxDot"
        />
      </div>
      <div>
        <p
          class="circle__text uppercase font-bold mt-3"
        >
          Krok 1
        </p>
        <p
          class="circle__text uppercase font-bold"
        >
          Wybierz Drzwi
        </p>
      </div>
    </div>
    <div
      class="stepper__line border-b border-mercury absolute z-10 left-half"
    />
  </div>
</DocumentFragment>
`;
